(feat) Added PaletteViewer to TestForm's StartScreen

tobitege opened the pull request:
Added PaletteViewer form to StartScreen and adjusted layout in TestForm.

Fixed index errror in PaletteMicrosoft365Blue to prevent runtime exception (out of bounds color index). 
Also fixed enum typos in comments, which are relevant for this tool!

Included Newtonsoft.Json package reference in project file.
Added manifest/config for HighDPI display support. 
StartScreen made slightly larger to not cut off buttons.

Fixes #2287 

Usage hints: specify path to Source folder; click "Add ALL" to fill grid with all compatible palettes; scroll down/across to find cells with error flags; "Missing value" means that the row's key is missing in the palette's array (can cause exceptions!).
Grid can be exported in CSV, JSON or XML format.

Development 100% directed by tobitege, 95% implemented by OpenAI's o3 model in close cooperation. 😁

giduac commented:
@tobitege, @lesandrog, @PWagner1, @Smurf-IV 

First testing the changes to Toolkit.

giduac commented:
Hi @tobitege 

Just thinking it over if JSon is really needed. That would save pulling in the Newtonsoft nuget into the repo. 
XML and Csv is probably enough for the export....

What's your take on this?

tobitege commented:
> Hi @tobitege
> 
> Just thinking it over if JSon is really needed. That would save pulling in the Newtonsoft nuget into the repo. XML and Csv is probably enough for the export....
> 
> What's your take on this?

No problem, I'll remove it completely.

giduac commented:
@tobitege 

Proposed additions please:

Add a KThemeCombo
![image](https://github.com/user-attachments/assets/199e8829-e4aa-410c-9995-555d5b50f65b)

Make KDGV rows autosize (makes it a bit slower but will show all multi line string in the cells.

```cs
        private void BuildBaseRows()
        {
            this.dataGridViewPalette.Columns.Clear();
            this.dataGridViewPalette.Rows.Clear();
            this.dataGridViewPalette.AutoSizeRowsMode = DataGridViewAutoSizeRowsMode.AllCells;
```

tobitege commented:
Fixed Json removal and row auto-sizing. Will add theme combo next.

giduac commented:
@tobitege 

**Little bug in reloading data:**
- start the checker
- press add palette
- press clear
- press add palette

Then api calls columns remains empty.


**And another one:**

Can we have a Environment.NewLine after each enum identifier so we'll get then nicely listed one at a time?

Current:
![image](https://github.com/user-attachments/assets/58e6f9a9-952f-475f-a29f-7acad0ae6b9d)

Desired layout:
![image](https://github.com/user-attachments/assets/3f03e386-5c99-4b7f-866b-891c1b996dec)

And the when all is correct we can set columns auto sizing for api calls col by adding.
```cs
// Column 2: API Call(s)
colApi.AutoSizeMode = DataGridViewAutoSizeColumnMode.AllCells;
```

giduac commented:
@tobitege 

We're getting there...

I think this can be disabled. I tried it and when turned the autosizing works correct.
Give it a try please.

```cs
private void DataGridViewPalette_ColumnAdded(object sender, System.Windows.Forms.DataGridViewColumnEventArgs e)
{
    // auto-size once then freeze
    //int preferred = e.Column.GetPreferredWidth(System.Windows.Forms.DataGridViewAutoSizeColumnMode.AllCells, true);
    //preferred = System.Math.Max(MinColumnWidth, System.Math.Min(preferred, 300));
    //e.Column.Width = preferred;
    //e.Column.AutoSizeMode = System.Windows.Forms.DataGridViewAutoSizeColumnMode.None;
    //e.Column.SortMode = System.Windows.Forms.DataGridViewColumnSortMode.NotSortable;
}

```

tobitege commented:
Tbh, I'd rather leave that in because it assures to disable sorting and keeps column width between 120 and 300, as sanity checks.

giduac commented:
@tobitege 

The please apply only if not col index 2.
```cs
private void DataGridViewPalette_ColumnAdded(object sender, System.Windows.Forms.DataGridViewColumnEventArgs e)
{
    // auto-size once then freeze
    int preferred = e.Column.GetPreferredWidth(System.Windows.Forms.DataGridViewAutoSizeColumnMode.AllCells, true);
    preferred = System.Math.Max( MinColumnWidth, System.Math.Min( preferred, 300 ) );
    e.Column.Width = preferred;
            
    if (e.Column.Index != 2)
    {
        e.Column.AutoSizeMode = System.Windows.Forms.DataGridViewAutoSizeColumnMode.None;
    }

    e.Column.SortMode = System.Windows.Forms.DataGridViewColumnSortMode.NotSortable;
}
```

tobitege commented:
Applied that column index check. Made theme selector combo wider so the lengthy names don't get truncated.

Smurf-IV commented:
Mt $0.02....
It would be nice if this was in the actual Palette designer repo.
Also, for the missing numbers to have some indication as to which are missing.

tobitege commented:
- API column is now frozen, which avoids paint issues while scrolling, too
- reverted value addition in PaletteMicrosoft365Blue.cs
- the changes in 2nd theme file should stay, they include enum name typo fixes and consistent formatting (helps with parsing tools like this!)
- fixed Cancel button positioning
This is as much time as I got for this now.

PWagner1 commented:
> Mt $0.02.... It would be nice if this was in the actual Palette designer repo. Also, for the missing numbers to have some indication as to which are missing.
> 
> But, This is all good work ;-)

@Smurf-IV 

I agree.

@tobitege Nice work, would you mind creating a separate PR for the Palette designer repo also? :)

tobitege commented:
Done :)

PWagner1 commented:
> > > Mt $0.02.... It would be nice if this was in the actual Palette designer repo. Also, for the missing numbers to have some indication as to which are missing.
> > > But, This is all good work ;-)
> > 
> > 
> > @Smurf-IV
> > I agree.
> > @tobitege Nice work, would you mind creating a separate PR for the Palette designer repo also? :)
> 
> Which repo is that exactly? 🤔

@tobitege 

It's this one https://github.com/[Krypton-Suite/Theme-Palettes](https://github.com/Krypton-Suite/Theme-Palettes)

giduac commented:
@tobitege 

So entering the source directory is mandatory for correct functioning and must be set before a palette can be added to the grid?

giduac commented:
@tobitege 

Should we make the path mandatory and always perform that relationship check ?

tobitege commented:
> @tobitege
> 
> Should we make the path mandatory and always perform that relationship check ?

Hmm... not sure if that's needed. IF someone compiles against NuGet instead of source, they could still use the minimal display version?

giduac commented:
@tobitege  

Over and out for today....

tobitege commented:
Added one more fix for high DPI handling across frameworks to make compiler happy.
